graph/pull.go

```diff
@@ -43,17 +43,19 @@ func (s *TagStore) Pull(image string, tag string, imagePullConfig *ImagePullConf
 		return err
 	}
 
-	c, err := s.poolAdd("pull", utils.ImageReference(repoInfo.LocalName, tag))
+	ps, err := s.poolAdd("pull", utils.ImageReference(repoInfo.LocalName, tag))
 	if err != nil {
-		if c != nil {
+		if ps.c != nil {
```

Comment:
Could we add a function to check and wait on c? This way ProgressStatus could isn't tied to the graph package.